[WIP] Design changes for CSV uploader

nathancrtr opened the pull request:
## Related Issue
- Addresses #3925 

## Changes Proposed
- Import the "upload guide" and "schema guide" from ReportStream
    - Updates links, copy, etc. as required

## Testing
- By the time you read this, this should be in `test`
 
## Screenshots / Demos
<img width="817" alt="Screen Shot 2022-06-29 at 3 18 09 PM" src="https://user-images.githubusercontent.com/27730981/176518339-da032e51-2abf-450b-97d5-4c7194da9a30.png">
<img width="903" alt="Screen Shot 2022-06-29 at 3 18 19 PM" src="https://user-images.githubusercontent.com/27730981/176518341-e00a866c-7688-42f5-8a86-dc8afa4118d6.png">
<img width="866" alt="Screen Shot 2022-06-29 at 3 18 41 PM" src="https://user-images.githubusercontent.com/27730981/176518342-d55de195-00bc-46c9-846f-9db8db3be074.png">

## Checklist for Author and Reviewer
- [ ] @vz3 there are _definitely_ some formatting fixes we will want here :)
    - More padding on the schema documentation page?
    - Uneven spacing above & below the top nav
    - Anything else? 
- [ ] @vz3 ReportStream provides an ["upload guide"](https://staging.reportstream.cdc.gov/getting-started/testing-facilities/csv-upload-guide) in addition to their schema documentation. Looking at that page, its information appears to be either ReportStream-specific or redundant with the inline instructions. Should we link to this as well?
- [ ] @lauraykerr does the copy look OK? Please double-check the links in the inline instructions to ensure they are what you intended

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                  Pilot program{" "}
+                </span>
+              </h2>
+              <p className="usa-intro text-base margin-bottom-4">
+                Step-by-step instructions and guidance for preparing and
+                uploading test results via a comma-separated values (CSV) file.
+              </p>
```
```suggestion
              <p className="usa-intro text-base margin-bottom-4">
                Step-by-step guidance for preparing and
                uploading test results via a standard comma-separated values (CSV) spreadsheet.
              </p>
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+              </p>
+              <p>
+                Use a simple online tool to submit a CSV file with a{" "}
+                <a
+                  href="/getting-started/testing-facilities/csv-schema"
+                  className="usa-link"
+                >
+                  standard schema
+                </a>
+                . Receive real-time validation and feedback on file format and
+                field values before submission.
+              </p>
+              <div className="usa-alert usa-alert--info margin-y-6">
+                <div className="usa-alert__body">
+                  <h3 className="usa-alert__heading font-body-md margin-top-05">
+                    About CSV upload
+                  </h3>
+                  This feature is currently being piloted in select
+                  jurisdictions with organizations or facilities that have
+                  existing Electronic Medical Record (EMR) systems. Pilot
+                  partners are selected by recommendation from jurisdictions.
+                  Find out if your jurisdiction is{" "}
+                  <a href="/how-it-works/where-were-live" className="usa-link">
+                    partnered
+                  </a>{" "}
+                  with SimpleReport and{" "}
+                  <a
+                    href={
+                      "mailto:" +
+                      "TODO" +
+                      "?subject=Getting started with SimpleReport"
+                    }
+                    className="usa-link"
+                  >
+                    contact us
+                  </a>{" "}
+                  to learn more.
+                </div>
+              </div>
+              <p>
```
Would recommend removing both of these sections, at least for the launch. There's a lot of content before "In this guide," and the initial users probably don't need information about what CSV uploader is or how to get signed up.

nathancrtr replied:
Done 👏 [afb0b7d](https://github.com/CDCgov/prime-simplereport/pull/3928/commits/afb0b7daff592ea23efc832618b88235dd03cac0)

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                    href="https://SimpleReport.cdc.gov/assets/csv/SimpleReport-StandardCSV-ExampleData-20220509.csv"
+                    className="usa-link"
+                  >
+                    SimpleReport standard CSV with example data
```
```suggestion
                    SimpleReport CSV template with example data
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+              <ol className="usa-process-list">
+                <li className="usa-process-list__item">
+                  <h4 className="usa-process-list__heading">
+                    Download the SimpleReport reference CSV and review the
+                    documentation
```
```suggestion
                    Download the SimpleReport CSV template and review the
                    documentation
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                    documentation
+                  </h4>
+                  <p className="margin-top-05">
+                    If your jurisdiction already has a set format for CSV,
+                    compare it to our{" "}
+                    <a
+                      href="https://SimpleReport.cdc.gov/assets/csv/SimpleReport-StandardCSV-ExampleData-20220509.csv"
+                      className="usa-link"
+                    >
+                      standard CSV file
```
```suggestion
                    If your jurisdiction already has a set format for CSV,
                    compare it to our{" "}
                    <a
                      href="https://SimpleReport.cdc.gov/assets/csv/SimpleReport-StandardCSV-ExampleData-20220509.csv"
                      className="usa-link"
                    >
                      CSV template file
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                  <h4 className="usa-process-list__heading">
+                    Enter values into your CSV file
+                  </h4>
+                  <p>
+                    Using the{" "}
+                    <a href="/getting-started/testing-facilities/csv-schema">
+                      {" "}
+                      CSV schema documentation
+                    </a>{" "}
+                    as a guide, enter properly formatted values in the relevant
+                    fields.
+                  </p>
```
```suggestion
                  <p>
                    Following the{" "}
                    <a href="/getting-started/testing-facilities/csv-schema">
                      {" "}
                      CSV template documentation
                    </a>{" "}
                   , enter properly formatted values in the relevant
                    fields.
                  </p>
```
Recommend renaming "CSV schema documentation" to "CSV template guide" across the board, since some people we tested with were unfamiliar with the term "schema."

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                    </li>
+                  </ul>
+                  <p>
+                    <strong>
+                      Optional: If you receive errors after clicking "Upload"
+                    </strong>
```
```suggestion
                    <strong>
                      Optional: Fix any errors
                    </strong>
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                      Optional: If you receive errors after clicking "Upload"
+                    </strong>
+                  </p>
+                  <p>
+                    SimpleReport validates all uploaded files against a{" "}
+                    <a
+                      href="/getting-started/testing-facilities/csv-schema"
+                      className="usa-link"
+                    >
+                      standard CSV schema
+                    </a>
+                    . If any errors are detected with a file's format or data,
+                    the application will alert you to specific changes you need
+                    to make before the file can be submitted.
+                  </p>
```
```suggestion
                  <p>
                    SimpleReport checks all uploaded files against a{" "}
                    <a
                      href="/getting-started/testing-facilities/csv-schema"
                      className="usa-link"
                    >
                      standard CSV schema, or data structure..
                    </a>
                    . If it detects any errors in a file's format or data,
                    the application will alert you to the specific changes you need
                    to make before submitting.
                  </p>
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                    the application will alert you to specific changes you need
+                    to make before the file can be submitted.
+                  </p>
+                  <p>Resolving errors with a CSV:</p>
+                  <ul>
+                    <li>
+                      Review the list of errors and recommended changes
+                      suggested by SimpleReport. Reference the{" "}
+                      <a
+                        href="/getting-started/testing-facilities/csv-schema"
+                        className="usa-link"
+                      >
+                        CSV schema documentation
+                      </a>{" "}
+                      again for any adjustments to column headers or values.
+                    </li>
```
```suggestion
                  <p>To resolve errors in a CSV:</p>
                  <ul>
                    <li>
                      Review the list of errors and recommended changes
                      suggested by SimpleReport. Reference the{" "}
                      <a
                        href="/getting-started/testing-facilities/csv-schema"
                        className="usa-link"
                      >
                        CSV template guide
                      </a>{" "}
                      again for any adjustments you need to make to column headers or values.
                    </li>
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvSchemaDocumentation.tsx:
```diff
@@ -0,0 +1,296 @@
+                </span>
+              </h2>
+              <p className="usa-intro text-base">
+                How to format data for submission to SimpleReport via CSV
+                upload.
```
```suggestion
                How to format data for submission to public health departments via a CSV
                spreadsheet.
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvSchemaDocumentation.tsx:
```diff
@@ -0,0 +1,296 @@
+                Updated: May 9, 2022
+              </span>
+              <h2 className="margin-top-0 ">
+                CSV schema documentation{" "}
```
```suggestion
                CSV template guide{" "}
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvSchemaDocumentation.tsx:
```diff
@@ -0,0 +1,296 @@
+                <a
+                  href="/getting-started/organizations-and-testing-facilities/where-does-simplereport-work"
+                  className="usa-link"
+                >
+                  partnered
+                </a>{" "}
+                with SimpleReport.{" "}
+              </p>
+
+              <div className="usa-alert usa-alert--info margin-y-6">
+                <div className="usa-alert__body">
+                  <h3 className="usa-alert__heading font-body-md margin-top-05">
+                    About CSV upload
+                  </h3>
+                  This documentation will help you prepare a file for CSV
+                  upload. This feature is currently being piloted in select
+                  jurisdictions with organizations or facilities that have
+                  existing Electronic Medical Record (EMR) systems. Pilot
+                  partners are selected by recommendation from jurisdictions.
+                  Find out if your jurisdiction is{" "}
+                  <a
+                    href="/getting-started/organizations-and-testing-facilities/where-does-simplereport-work"
+                    className="usa-link"
+                  >
+                    partnered
+                  </a>{" "}
+                  with SimpleReport and{" "}
+                  <a
+                    href={
+                      "mailto:" +
+                      "support@simplereport.gov" +
+                      "?subject=Getting started with SimpleReport"
+                    }
+                    className="usa-link"
+                  >
+                    contact us
+                  </a>{" "}
+                  to learn more.
+                </div>
+              </div>
```
Similar to the other page, recommend moving the blurb about the schema to the top of the next section and removing the "About CSV upload."

lauraykerr commented on frontend/src/app/testResults/uploads/CsvSchemaDocumentation.tsx:
```diff
@@ -0,0 +1,296 @@
+                id="formatting-guidelines"
+                className="font-body-lg margin-y-1"
+              >
+                General formatting guidelines
+              </h3>
+
```
```suggestion
                General formatting guidelines
              </h3>
              <p>
                The SimpleReport standard CSV template is a blend of the
                Department of Health and Human Science's (HHS){" "}
                <a
                  href="https://www.hhs.gov/coronavirus/testing/covid-19-diagnostic-data-reporting/index.html"
                  target="_blank"
                  rel="noreferrer noopener"
                  className="usa-link"
                >
                  requirements for COVID-19 test data
                </a>{" "}
                as well as those of numerous jurisdictions. This standard schema
                will be accepted by state, tribal, local, or territorial (STLT)
                 {" "}
                <a
                  href="/getting-started/organizations-and-testing-facilities/where-does-simplereport-work"
                  className="usa-link"
                >health departments
                  partnered
                with SimpleReport.</a>{" "}
              </p>

```

lauraykerr commented:
@nathancrtr Added some comments and suggestions directly in the guidance. The other changes I'd recommend are: 

1. Changing "CSV schema documentation" to "CSV template guide" across the board 
2. Changing "Test results" to "See test results" in sub nav to mirror "Upload spreadsheet" and "View upload history"

Let me know if you have questions or if there's an easier way to suggest changes!

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -143,91 +147,123 @@ const Uploads = () => {
+              <ul className="usa-list">
+                <li>
+                  <a href="https://reportstream.cdc.gov/assets/csv/ReportStream-StandardCSV-ExampleData-20220509.csv">
+                    Download the spreadsheet template
```
```suggestion
                    Download the spreadsheet template and open it in Excel or another spreadsheet application
```
Folding in the gray text currently above the upload feature

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -143,91 +147,123 @@ const Uploads = () => {
+                <li>Save your spreadsheet as a CSV file and upload it here</li>
+              </ul>
+              <p>
+                For more information about the spreadsheet, visit our{" "}
+                <LinkWithQuery to="/results/upload/schema">
+                  data template guide.
```
```suggestion
                For more information about this process, visit our{" "}
                <LinkWithQuery to="/results/upload/schema">
                  data template guide or CSV upload guide.
```

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -143,91 +147,123 @@ const Uploads = () => {
+                <li>Save your spreadsheet as a CSV file and upload it here</li>
+              </ul>
+              <p>
+                For more information about the spreadsheet, visit our{" "}
+                <LinkWithQuery to="/results/upload/schema">
+                  data template guide.
+                </LinkWithQuery>
```
```suggestion
                For more information about this process, visit our{" "}
                <LinkWithQuery to="/results/upload/schema">
                  data template guide
                </LinkWithQuery> or upload guide.
```
Will need to add link to "upload guide," but not totally sure how to

nathancrtr replied:
Done in [afb0b7d](https://github.com/CDCgov/prime-simplereport/pull/3928/commits/afb0b7daff592ea23efc832618b88235dd03cac0) 👍 the upload guide is the `CsvUploadGuide.tsx` file that you reviewed as a part of this PR so we should be good

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -143,91 +147,123 @@ const Uploads = () => {
-              {!isSubmitting && <span>Upload</span>}
-            </Button>
+              <FormGroup className="margin-bottom-3">
+                <Label
+                  className="font-sans-xs usa-hint"
+                  id="upload-csv-input-label"
+                  htmlFor="upload-csv-input"
+                >
+                  Upload your test results as a spreadsheet. You can create a
+                  .csv file in any spreadsheet application like Excel or Google
+                  Sheets
+                </Label>
```
I'd recommend removing this for now, since most of the information is covered in the steps above (added the spreadsheet application piece to the first bullet).

nathancrtr replied:
Makes sense! Removed in [5f3dcc5](https://github.com/CDCgov/prime-simplereport/pull/3928/commits/5f3dcc516c9a34235bf87ebfd2d9aacecbde6921).

nathancrtr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                id="upload-a-csv"
+                className="font-body-lg border-top-1px border-ink margin-top-8 margin-bottom-6 padding-top-1"
+              >
+                Uploading a CSV file
```
@lauraykerr what do you think about this section (image attached)?

<img width="509" alt="Screen Shot 2022-06-30 at 12 00 11 PM" src="https://user-images.githubusercontent.com/27730981/176724264-1cad2b9c-6a9a-4f11-9768-2ca2a1c4da15.png">

On ReportStream, these documents are located on the static site and are available to unauthenticated viewers. At least for now, though, anyone using the uploader in SimpleReport will already have logged in, navigated to the uploader, etc.

lauraykerr replied:
@nathancrtr Good call out — I think we can scrap step #1. We also probably want to edit #2 (now #1) to say: 
**Navigate to "Upload spreadsheet**
You can find "Upload spreadsheet" under "Results" in the main navigation at the top of the page.

lauraykerr commented on frontend/src/app/testResults/uploads/CsvUploadGuide.tsx:
```diff
@@ -0,0 +1,271 @@
+                Uploading a CSV file
+              </h3>
+              <ol className="usa-process-list">
+                <li className="usa-process-list__item">
+                  <h4 className="usa-process-list__heading">
+                    Log in to SimpleReport
+                  </h4>
+                  <ul className="margin-top-2">
+                    <li>
+                      Go to{" "}
+                      <a href="/login">https://SimpleReport.cdc.gov/login</a>
+                    </li>
+                    <li>Enter username</li>
+                    <li>Enter password</li>
+                    <li>Click "Sign in"</li>
+                  </ul>
+                </li>
```
```suggestion
         
```
Remove, since all users will be logged in already.

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -143,91 +147,123 @@ const Uploads = () => {
+                  match the template
+                </li>
+                <li>
+                  Make sure you include data for all fields marked “required” in
+                  row 2
```
```suggestion
                  Make sure you include data for all fields marked “required” in
                  the CSV template guide
```
Keeping the template unchanged for now, so changing this language.

nathancrtr commented on frontend/src/app/testResults/submissions/Submissions.tsx:
```diff
@@ -11,8 +11,11 @@ import {
 import { LinkWithQuery } from "../../commonComponents/LinkWithQuery";
+import { useDocumentTitle } from "../../utils/hooks";
 
 const Submissions = () => {
+  useDocumentTitle("View upload history");
+
```
Good question... maybe I'll keep this as `Submissions` since it matches the URL path?

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -140,98 +144,120 @@ const Uploads = () => {
+          <h2>Upload your results</h2>
+        </div>
+        <div className="usa-card__body padding-y-2 maxw-prose">
+          <p>
+            Add and report results in bulk using a comma-separated values (CSV)
+            spreadsheet. To upload your spreadsheet:
+          </p>
+          <ul className="usa-list">
+            <li>
+              <a href="https://reportstream.cdc.gov/assets/csv/ReportStream-StandardCSV-ExampleData-20220509.csv">
+                Download the spreadsheet template
+              </a>{" "}
+              and open it in Excel or another spreadsheet application
+            </li>
+            <li>
+              Add your data to the relevant fields in the template, or if you’re
+              using your own spreadsheet, format column headers to match the
+              template
+            </li>
+            <li>
+              Make sure you include data for all fields marked “required” in row
+              2
+            </li>
+            <li>Once you’ve entered your data, delete all example data</li>
+            <li>Save your spreadsheet as a CSV file and upload it here</li>
+          </ul>
+          <p>
+            For more information about this process, visit our{" "}
+            <LinkWithQuery to="/results/upload/schema">
+              data template guide
+            </LinkWithQuery>{" "}
+            or{" "}
+            <LinkWithQuery to="/results/upload/guide">
+              upload guide.
+            </LinkWithQuery>
+          </p>
```
```suggestion
          <p>
           Report results in bulk using a comma-separated values (CSV)
            spreadsheet. To upload your spreadsheet:
          </p>
          <ol className="usa-list">
            <li>
              <a href="https://reportstream.cdc.gov/assets/csv/ReportStream-StandardCSV-ExampleData-20220509.csv">
                Download the spreadsheet template
              </a>{" "}
            </li>
            <li>
              Format and add your data following the template and upload guide
            </li>
            <li>
              Save your spreadsheet as a CSV
            </li>
            <li>Upload your CSV here</li>
          </ol>
          <p>
            For more information about preparing your spreadsheet, visit our{" "}
            <LinkWithQuery to="/results/upload/schema">
              spreadsheet upload guide
            </LinkWithQuery>{" "}
          </p>
```

nathancrtr replied:
@lauraykerr it looks like this change removes the link to the "upload guide" at `/results/upload/guide`. Now it only has the link to the CSV schema. Do we want to preserve the path to the guide here?

lauraykerr replied:
@nathancrtr With design feedback, we decided to consolidate the upload guide and schema documentation (see suggestions for the new section in the schema file).

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.test.tsx:
```diff
@@ -41,10 +41,7 @@ describe("Uploads", () => {
-    expect(
-      await screen.findByText("Upload your COVID-19 lab results as a .csv.")
-    ).toBeInTheDocument();
+    expect(await screen.findByText("Upload your results")).toBeInTheDocument();
```
```suggestion
    expect(await screen.findByText("Upload your CSV")).toBeInTheDocument();
```

lauraykerr replied:
Correct, thank you!

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.test.tsx:
```diff
@@ -41,10 +41,7 @@ describe("Uploads", () => {
-      await screen.findByText("Upload your COVID-19 lab results as a .csv.")
-    ).toBeInTheDocument();
+    expect(await screen.findByText("Upload your results")).toBeInTheDocument();
     expect(await screen.findByText("Drag file here or")).toBeInTheDocument();
```
```suggestion
    expect(await screen.findByText("Drag CSV here or")).toBeInTheDocument();
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvSchemaDocumentation.tsx:
```diff
@@ -0,0 +1,252 @@
+                href="/getting-started/testing-facilities/csv-schema"
+                className="usa-link"
+              >
+                Standard schema with example data
```
```suggestion
                Standard CSV template with example data
```

lauraykerr commented on frontend/src/app/testResults/uploads/CsvSchemaDocumentation.tsx:
```diff
@@ -0,0 +1,252 @@
+            </li>
+          </ul>
+        </section>
+
```
```suggestion
 <section className="border-top-1px border-ink margin-top-9">
          <h3 id="formatting-guidelines" className="font-body-lg margin-y-1">
          Preparing and uploading a spreadsheet
          </h3>
          <ol className="usa-process-list">
            <li className="usa-process-list__item">
              <h4 className="usa-process-list__heading">
                Create or export your spreadsheet
              </h4>
              <p className="margin-top-05">
               If your organization already uses a set spreadsheet format for results, you may have to adjust it to match the SimpleReport template. If you don’t have one, use the {" "}
                <a
                  href="https://SimpleReport.cdc.gov/assets/csv/SimpleReport-StandardCSV-ExampleData-20220509.csv"
                  className="usa-link"
                >
                  spreadsheet template file
                </a>{" "} as a starting point. 
              </p>
            </li>
            <li className="usa-process-list__item">
              <h4 className="usa-process-list__heading">
                Format using the SimpleReport template
              </h4>
              <p>
                In your spreadsheet, include all column headers in the spreadsheet template and guidelines below, with no extras. Copy column header names exactly. Be sure to include every column in the template, even if you don’t have data for every field. 
              </p>
            </li>
            <li className="usa-process-list__item">
              <h4 className="usa-process-list__heading">
                Enter your data</h4>
              <p>
               Following the spreadsheet guidelines below, enter properly formatted values in the relevant fields. Some fields require data, while others don’t.
              </p>
            </li>
            <li className="usa-process-list__item">
              <h4 className="usa-process-list__heading">Export or save your data</h4>
              <p>Make sure your spreadsheet is in a CSV format. SimpleReport does’t accept .XLS, .XLXS, or other formats.</p>
            </li>
            <li className="usa-process-list__item">
              <h4 className="usa-process-list__heading">Use the uploader on SimpleReport</h4>
              <p>Visit the “[Upload spreadsheet](url)” tab under “Results” in the main SimpleReport navigation. Select your CSV by dragging the file from a folder to the upload area, or browse your computer to find and open it. Once you click Upload your CSV, SimpleReport will check your file to see if it matches the template. If it accepts the file, you’ll see a confirmation message.</p>
              <h4 className="usa-process-list__heading">Fix any errors</h4>
              <p>If SimpleReport finds any errors in the spreadsheet formatting or data, it will recommend how to fix them. Once you’ve made the recommended changes in your spreadsheet, save it, and then upload it again. 
</p>
            </li>
          </ol>
        </section>
```
Note: will need to add proper link on step 5, "Upload spreadsheet"

lauraykerr replied:
This update simplifies the upload guide and adds it to the schema documentation. We won't need the documentation previously known as "CSV upload guide," but should rename this page "Spreadsheet upload guide."

cc @nathancrtr @vz3

nathancrtr replied:
Love it, I've been thinking that the "upload guide" was a little redundant 👍

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -140,98 +144,111 @@ const Uploads = () => {
-              {!isSubmitting && <span>Upload</span>}
-            </Button>
-          </div>
+            {!isSubmitting && <span>Upload your spreadsheet</span>}
```
```suggestion
            {!isSubmitting && <span>Upload your CSV</span>}
```

lauraykerr commented on frontend/src/app/testResults/uploads/Uploads.tsx:
```diff
@@ -140,98 +144,110 @@ const Uploads = () => {
+          </ol>
+          <p>
+            For more information about preparing your spreadsheet, visit our{" "}
+            <LinkWithQuery to="/results/upload/guide">
+              spreadsheet upload guide
+            </LinkWithQuery>{" "}
```
```suggestion
            <LinkWithQuery to="/results/upload/guide">
             <b> spreadsheet upload guide</b>
            </LinkWithQuery>{" "}
```

lauraykerr commented:
@nathancrtr It's looking great on `test`. A couple copy notes for the guide: 
- Can we add anchor link to the “Preparing and uploading a spreadsheet” section under “In this guide”?
- Does the "Updated" date change automatically? If not, wondering if we want to remove for now. It currently says May 9, 2022. 
- Added a suggested to bold the link to the guide in the uploader instructions

@vz3 One note more in the design wheelhouse, but with content implications: 
- Realizing that by adding the "process list" class to the upload steps, the headers in this section (h3 and h4) are the same sizes

nathancrtr commented:
@lauraykerr thanks for the comments! Added the following in [99892cd](https://github.com/CDCgov/prime-simplereport/pull/3928/commits/99892cdf47bce566b9d3793d7d6970c86a28086f):

* Added the anchor link
* Removed the `Updated` line as it is not updated automatically
* Bolded the link to the spreadsheet upload guide

<img width="664" alt="Screen Shot 2022-07-08 at 12 09 00 PM" src="https://user-images.githubusercontent.com/27730981/178030491-5360eedc-3d3e-4cde-83f5-f790e84e9420.png">
<img width="634" alt="Screen Shot 2022-07-08 at 12 09 18 PM" src="https://user-images.githubusercontent.com/27730981/178030492-4e74679e-272b-43d2-87b4-096f3d730c2d.png">

lauraykerr commented:
@nathancrtr Suggested switching order of "Preparing and uploading spreadsheet" in "In this guide," otherwise LGTM, thank you!

nathancrtr commented:
> @nathancrtr Suggested switching order of "Preparing and uploading spreadsheet" in "In this guide," otherwise LGTM, thank you!

Done 👏 will await your approval and get this in asap!

lauraykerr reviewed (dismissed):
LGTM!

vz3 commented:
IMO we need a "Return to" link at the top of the guide page. I think it should bring the user back to the "Upload spreadsheet" page.

@lauraykerr how about `< Return to Upload spreadsheet`

vz3 commented:
Also we need a period after "For more information about preparing your spreadsheet, visit our spreadsheet upload guide." @nathancrtr

lauraykerr commented:
@vz3 Good point on the "return to" link. My preference would be either `< Return to **Upload spreadsheet`** (bold, highlighting that it's a feature), or `< Return to upload spreadsheet` (lowercase, highlighting that it's an action). What do you think?

nathancrtr commented:
@vz3 good call, added that period in.

<img width="616" alt="Screen Shot 2022-07-08 at 4 46 27 PM" src="https://user-images.githubusercontent.com/27730981/178067946-d3849d20-2962-440e-8183-17966739e9d0.png">

@lauraykerr I went ahead and pushed your latter (lowercase) suggestion but can change it to anything else if we decide

<img width="610" alt="Screen Shot 2022-07-08 at 4 46 17 PM" src="https://user-images.githubusercontent.com/27730981/178067944-8176ca14-8c20-4f7c-b97c-e7271221d47a.png">

lauraykerr commented:
@nathancrtr It's look great on `test`! A few things, organized by page: 

**Upload spreadsheet**
- Download spreadsheet in uploader downloads a ReportStream file — is this right, or do we have a SR version? 

**View upload history**
- Can we use the same copy as test results page for the date range inputs: 
    - “Submitted (Start Range)” > “Date range (start)”
    - “Submitted (End Range)” > “Date range (end)”

**Spreadsheet upload guide**
- The “Return to upload spreadsheet” link brought me to the Upload history page (should go to Upload spreadsheet)
- On step 1, spreadsheet template links to 404 — probably my bad when I added this section
- On step 5, we need to add link to Upload spreadsheet on “[Upload spreadsheet](url)” 

**Conduct tests**
- As an attempt to cut down on this ticket, I created a [new ticket for adding a link to upload spreadsheet](https://github.com/CDCgov/prime-simplereport/issues/3939#issuecomment-1178082145) on the empty card in "Conduct tests," but promptly failed to add you to it. Sorry about that!

lauraykerr commented:
@nathancrtr looks good to me, thank you! 

Noting that at some point should do an audit of how we're using quotes vs. bold in text across the app/site (cc @vz3).

lauraykerr reviewed (approved):
LGTM!

vz3 commented:
Looks good here!
